File: src/models/sessionregistration.js
"use strict";
const { Model } = require("sequelize");
module.exports = (sequelize, DataTypes) => {
  class SessionRegistration extends Model {
    static associate(models) {
      this.belongsTo(models.User, { foreignKey: "user_id" });
      this.belongsTo(models.Session, { foreignKey: "session_id" });
    }
  }
  SessionRegistration.init(
    {
      id: {
        allowNull: false,
        autoIncrement: true,
        primaryKey: true,
        type: DataTypes.INTEGER,
      },
      user_id: {
        type: DataTypes.UUID,
        allowNull: false,
        references: {
          model: "Users",
          key: "id",
        },
      },
      session_id: {
        type: DataTypes.INTEGER,
        allowNull: false,
      },
    },
    {
      sequelize,
      modelName: "SessionRegistration",
    }
  );
  return SessionRegistration;
};

File: internal/models/domain/SessionRegistration.go
package domain

import (
	"time"

	"gorm.io/gorm"
)

type SessionRegistration struct {
	gorm.Model
	ID        uint    `gorm:"primaryKey"`
	SessionID uint    `gorm:"not null"`
	Session   Session `gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`

	UserID uint `gorm:"not null"`
	User   User `gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`

	CreatedAt time.Time
	UpdatedAt time.Time
}
